### ISSUE TITLE
Why the R2 coverage is extremely low, while R1 performed normally?

### ISSUE THREAD

**Issue by huiyuan-qi** [FelixKrueger/Bismark#395]
Dear Felix,

I got trimmed, synchronized but mixed orientations R1 and R2 files in an amplicon bisulfite sequencing experiment, and wanted to use pair-end alignment in bismark. 
R1 and R2 have similar quality and indeed got a similar result when I ran single-end with each file. However, in the pair end result, the one that appeared firstly in the code, here is the R1, got relatively high coverage while the later one showed quite low coverage (referring to the M-bias.txt).

Do you know what is the reason and how to fix it?
Thanks a lot!

My code:
bismark GENOME --non-directional -1 R1.fq -2 R2.fq
bismark_methylation_extractor --bedGraph --gzip XX.bam

[R1&R2&referenceseq.zip](https://github.com/FelixKrueger/Bismark/files/5577878/R1.R2.referenceseq.zip)
[M-bias.txt](https://github.com/FelixKrueger/Bismark/files/5577885/M-bias.txt)

**Comment by FelixKrueger** [comment]
My guess would be that your amplicons are so short that R1 and R2 are overlapping, and therefore R2 methylation calls will get removed due to the no-overlap detection in the methylation extractor. Is that possible? If you run

```
bismark2report
```

once the run is complete, you should get nice HTML report that will show you the number of methylation calls per read. As a control, you could test the option `--include_overlap` or similar, in which case the number of calls should be higher again (I wouldn't recommend to use these additional calls as they are just PCR generated. 

Let me know how you get on.

**Comment by huiyuan-qi** [comment]
Thank you for the rapid reply! After using --include_overlap, the R2 got similar calls with R1. I would not use the additional calls as you suggested.
Another thing I am confusing is that the alignment efficiency of my library is not really high. The alignment efficiency of pair-end, and the single-end of R1 and R2 were all about 30% in non-directional mode.  Do you think it is the problem of my code or my experiment? And will non-directional be proper? (In directional mode the efficiency reduced by half and the coverage will be not enough in post-analysis, I think.)

**Comment by FelixKrueger** [comment]
It is a little difficult to say much about mapping efficiencies without seeing the data as such. Did you align the data just to some sequences, or to a full genome? Do you see a lot of sequences in the category "Genomic sequence could not be extracted"? If this were the case, then extending the sequences with `NN` even at the end of sequence would help (I saw that you included this it at the very start). 

If you could send me a test file (raw data ideally) I could take a quick look.

**Comment by huiyuan-qi** [comment]
Thank you! 
 I aligned my data to only to some sequences in the file referenceseq.fasta in the previous zip file. I did added NN at the end of the longest seq 'amp8' to avoid "Genomic sequence could not be extracted" and did not get this error message. By the way, the reference sequences are from cat genome (felis catus 9.0).

I found that after the adjustment of the --score_min of bowtie2 options, I could upper the mapping efficiency to 40%, though still not high. So maybe the problem is my library was not good enough? **I also do not sure what is the proper set for  -L -N --score_min**... If you could give me some suggestion that will be really helpful!

Here below is the raw fastq files.[raw fastq.zip](https://github.com/FelixKrueger/Bismark/files/5582572/raw.fastq.zip)

Thank you so much for your kind help!

**Comment by FelixKrueger** [comment]
If you don't have a specific reason, I would normally leave `-L -N` etc at the defaults. For `--score_min` I would probably allow a few more mismatches (such as setting it to `L,0,-0.4`) for genomes that are not quite as polished as the or mouse ones are.

I have just done a few tests with your data, and have to say it isn't all that conclusive... I found that the mapping efficiency (against the older Felix cattus genome 8.0) was ~20% in default mode (Trim Galore, Bismark with `--non_directional`), ~30% with a few more mismatches. And ~60% with `--local` mode. As expected, the were ~10 locations in the genome with a high number of reads, and quite a few off target locations (a lot more in `--local` mode, which is kind of expected as well.

Since there was quite some biased sequence a the start of your reads (and a few Ns), I experimented with `--clip_r1 --clip_r2`, as well as on the 3' end. This seemed to increase the mapping efficiency to ~42% in end-to-end mode. So there seems to be something inhibitory mostly in the 5' portions of reads, which you might want to investigate further. Single end alignments with clipped reads even reached 53%.

There were no other contaminations found in the sample, and the adapter contamination is also only quite moderate... I hope this will still give you something to experiment with?
